lldb/cmake/modules/LLDBConfig.cmake

```diff
@@ -57,7 +57,17 @@ add_optional_dependency(LLDB_ENABLE_CURSES "Enable curses support in LLDB" Curse
 add_optional_dependency(LLDB_ENABLE_LZMA "Enable LZMA compression support in LLDB" LibLZMA LIBLZMA_FOUND)
 add_optional_dependency(LLDB_ENABLE_LUA "Enable Lua scripting support in LLDB" LuaAndSwig LUAANDSWIG_FOUND)
 add_optional_dependency(LLDB_ENABLE_PYTHON "Enable Python scripting support in LLDB" PythonAndSwig PYTHONANDSWIG_FOUND)
-add_optional_dependency(LLDB_ENABLE_LIBXML2 "Enable Libxml 2 support in LLDB" LibXml2 LIBXML2_FOUND VERSION 2.8)
+# LibXml2 is required if LLDB_ENABLE_LIBXML2 is ON or AUTO. The LibXml2 package
+# search is not setup properly to support the CONFIG mode search. Furthermore,
```

Comment:
"The LibXml2 package search is not setup properly to support the CONFIG mode search."

Could you clarify who has not set this up properly? LLDB's CMake configuration or libxml2 itself?

Reply:
I believe, the issue is on the CMake side :( The documentation states that a `find_package` is supposed to search for the `CONFIG` and `MODULE` mode. This is now what I'm seeing with libxml2. It only searches in `MODULE` mode.

The CONFIG mode works completely once invoked. The problem is I'm worried that not everyone is building libxml2 with CMake and the `CONFIG` mode works only for those that are using CMake.

I can rewrite the comment.

Reply:
> In this case, by default, CMake first tries Module mode by searching for a Find<pkg>.cmake module. If it fails, CMake then searches for the package using Config mode.

- https://cmake.org/cmake/help/latest/variable/CMAKE_FIND_PACKAGE_PREFER_CONFIG.html

The `FindLibXml2.cmake` module exists, so it's using that. If you use `CMAKE_FIND_PACKAGE_PREFER_CONFIG=YES`, it should go with your config file.

Reply:
That seems reasonable. I'll trim this down to the removal of the additional include paths then.